istioctl/pkg/util/formatting/formatter_test.go

```diff
@@ -147,8 +147,8 @@ func TestFormatter_PrintEmpty(t *testing.T) {
 	g.Expect(logOutput).To(Equal(""))
 
 	jsonOutput, _ := Print(msgs, JSONFormat, false)
-	g.Expect(jsonOutput).To(Equal("[]"))
+	g.Expect(jsonOutput).To(Equal(""))
```

Comment:
I don't understand? `[]` is valid JSON for an empty list. The result is an empty list. Why do we not want to output the result?

Reply:
Yes it's valid JSON format. What I was trying to do is to make all three output formats have the same output results if there is no message needed to be printed. The original results will be below:
```
➜  testIstio cat bad.yaml | istioctl analyze --use-kube=false -olog -
Error(s) adding files: 1 error occurred:
	* errors parsing content "-": 1 error occurred:
	* error processing -[0]: failed converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context




No validation issues found when analyzing namespace: default (but 1 file could not be parsed).
Error: Some files couldn't be parsed.
➜  testIstio cat bad.yaml | istioctl analyze --use-kube=false -ojson -
Error(s) adding files: 1 error occurred:
	* errors parsing content "-": 1 error occurred:
	* error processing -[0]: failed converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context



[]
No validation issues found when analyzing namespace: default (but 1 file could not be parsed).
➜  testIstio cat bad.yaml | istioctl analyze --use-kube=false -oyaml -
Error(s) adding files: 1 error occurred:
	* errors parsing content "-": 1 error occurred:
	* error processing -[0]: failed converting YAML to JSON: yaml: line 2: mapping values are not allowed in this context



[]

No validation issues found when analyzing namespace: default (but 1 file could not be parsed).
```
The JSON format will print an extra string `[]` and the YAML format will print `[]\n`.

Reply:
> What I was trying to do is to make all three output formats have the same output results if there is no message needed to be printed.

I don't think this is a good goal. JSON output is for machines. Machines like the `[]`.

Reply:
Alright I see. Thanks